oauth-client/src/main/java/io/strimzi/kafka/oauth/client/JaasClientOauthLoginCallbackHandler.java

```diff
@@ -157,6 +158,38 @@ public void configure(Map<String, ?> configs, String saslMechanism, List<AppConf
         }
     }
 
+    private void checkConfiguration() {
```

Comment:
Should the log messages be in terms of the configuration parameters (e.g. `oauth.client.id` etc), rather than calling them by "client id"?

Reply:
Naming the precise config attributes has been a bit of a mess. Initially during OAuth library development I used the `oauth.client.id` format. Then switched to `OAUTH_CLIENT_ID` as I added support for ENV vars. But then as documentation grew, most of the references in documentation used `oauth.client.id` format.

It would be best to have another PR, a general cleanup where we convert all logged references from ENV var format to JAAS attribute format. But then, that could break backword compatibility for situations where tests, or alerting system filters rely on parsing logs, and finding the existing logging strings.

Reply:
Test should be easy to fix. Would alerting system filters really be parsing logs at that level of detail, and even for _configuration_ errors, rather than errors later on at runtime? Did we ever say that logging was a supported API offerings compatibility guarantees? For these reasons I think it would be worth doing that other PR.

Reply:
There's no way of knowing what people are doing, so the safest bet is always to assume the worst :)

Even I in the configuration tests rely on parsing the log. It may not be the prettiest approach, but it requires zero mocking and it leaves zero doubt if the configuration made it all the way to the listener, for example. You also test that debug logging actually outputs the configuration you expect it should output, so you're testing two things at once.
But yes, tests would be easy to fix. Maybe slightly more difficult in some Strimzi Operator system tests if they also use the same approach to testing.

We certainly don't make any promise that we won't change the logging. That other PR sure makes a lot of sense to clean the logging up.